Makefile

```diff
@@ -35,26 +44,48 @@ clean: \
 check-syntax-errors: .
 	@$(call help,$@:)
 	@python -m compileall -q $?
+	@$(call pass,$@:)
+
+## Run linters to check for syntax and style errors.
+check-style: \
+	check-style-all \
+	check-style-doc-commons \
+	check-style-doc-types
+	@$(call pass,$@:)
 
 ## Run linters to check for syntax and style errors.
-check-style: $(shell git ls-files "*.py")
+check-style-all:
 	@$(call help,$@:)
-	@flake8 $?
+	@flake8 `git ls-files | grep "\.py$$"`
 
-## Run code formatters to correct style errors.
-format-style: $(shell git ls-files "*.py")
+## Run linters to check for syntax and style errors.
+check-style-doc-%:
 	@$(call help,$@:)
-	@autopep8 $?
+	@flake8 --select=D101,D102,D103,DAR openfisca_core/$*
+	@pylint --enable=classes,exceptions,imports,miscellaneous,refactoring --disable=W0201,W0231 openfisca_core/$*
+
+## Run static type checkers for type errors.
+check-types: \
+	check-types-all \
+	check-types-strict-commons \
+	check-types-strict-types
```

Comment:
The difference between these targets is not clear, and the `-all` is misleading, as explained above for `check-style` :wink:

Reply:
Yes, I agree, what I propose is a specific task for maintainers, as discussed. I'll do that.

Reply:
Fixed.